src/hlsBinaries.ts

```diff
@@ -224,15 +289,8 @@ export async function downloadHaskellLanguageServer(
   const binaryDest = path.join(context.globalStoragePath, serverName);
 
   const title = `Downloading haskell-language-server ${release.tag_name} for GHC ${ghcVersion}`;
-  try {
-    await downloadFile(title, asset.browser_download_url, binaryDest);
-    return binaryDest;
-  } catch (e) {
-    if (e instanceof Error) {
-      window.showErrorMessage(e.message);
-    }
-    return null;
-  }
+  await downloadFile(title, asset.browser_download_url, binaryDest);
```

Comment:
Are the errors from `downloadFile` being caught or displayed anywhere? `downloadFile` can throw an error if the internet connection goes during the download etc which would be nice to display to the user

Reply:
They should get caught here in [activateServerForFolder](https://github.com/haskell/vscode-haskell/blob/65a55b6c99b5c7824675a5b52914a7e22490deec/src/extension.ts#L161-L165). Let me try to intentionally fail the download real quick.

Reply:
An error shows up at that simply says `getaddrinfo EAI_AGAIN github.com`, probably want to improve the message a bit here.

Reply:
Now says something like `Failed to download <url of asset>: getaddrinfo EAI_AGAIN github.com`, which should be more clear.